src/esp32/motor.rs

```diff
@@ -328,3 +379,67 @@ where
         Ok(Some(prost_types::Struct { fields: bt }))
     }
 }
+
+/// Below is a first attempt as an approach to runtime configuration, the problem raise with runtime configuration is
+/// enforcing single instance of any peripheral at any point in the program. For example say you have a Motor that uses pins 33,34 and 35 and
+/// a AnalogReader that uses pin 35. This situation is wrong since two objects own pin 35. In embedded rust this is avoided by having any hardware
+/// peripherals be singleton and leveraging the borrow checker so that single ownership rules are enforced. When dealing with runtime configuration,
+/// the borrow checker cannot help us. We can however follow the singleton approach and wrap peripherals into options that will be 'taken out' when
+/// something needs an hardware component. Following is an implementation of the proposed approach, a significant limitation is that the hardware can
+/// only be taken once and can never be returned.
```

Comment:
Yes we could use Rc/Arc pointers for that, the big question remains how do we value flexibility with added overhead. This would make a lot of sense if we want to support full live reconfiguration like in RDK, we could also accept that reconfiguration requires a reboot!